Refactor plugins mcps quickstart docs for engineers (#102)
griddynamics/rosetta: PLUGINS.md

# Plugins

Rosetta plugins bundle the bootstrap rule, skills, agents, workflows, and other instructions directly into your IDE. The agent loads them locally — no live connection to Rosetta is needed at request time.

Every plugin supports two installation methods:

- **Marketplace** — managed install from a plugin marketplace. Easier; preferred when available.
- **Standalone** — manual zip extraction into your repo. For agents without a marketplace path, or environments that block external marketplaces.

> [!CAUTION]
> You must receive prior approval from your manager and company to use Rosetta.

> [!WARNING]

> > Use **Sonnet 4.6**, **GPT-5.3-codex-medium**, **gemini-3.1-pro**, or better. Avoid Auto.

Author: ```suggestion
> Use **Sonnet 4.6**, **gpt-5.4-medium**, **gemini-3.1-pro** or better models. Avoid Auto model selection.
```

> [!NOTE]

> > Plugins are pre-release.

Author: ```suggestion
```

## Step 1: Install Plugin

<details>
<summary><b>Claude Code</b></summary>

### Claude Code

#### Marketplace

```sh
claude plugin marketplace add griddynamics/rosetta
claude plugin install rosetta@rosetta
```
</details>

<details>
<summary><b>Cursor</b></summary>

### Cursor

> [!NOTE]
> Cursor also sees plugins installed via Claude Code. If you've already installed via `claude plugin install`, do **not** install again in Cursor — the same content would be duplicated in Cursor's context.

> #### Marketplace

Author: AI mixed everything, original text was better, but lines 29 and 31 must be exchanged and it could be rephrased. Convert to action first sentences.

Publish to marketplace
Alternative: install to claude code
Warning: do not install both!

Reviewer: fixed

If your organization uses Cursor Team Marketplace, you can publish the Rosetta plugin to your company's internal marketplace and make it available to your team.

For more information, see the Cursor documentation:
https://cursor.com/docs/plugins#team-marketplaces

Rosetta can also be installed in Claude Code. Plugins installed in Claude Code are automatically available in Cursor.

The Rosetta repository provides both a Cursor plugin and marketplace integration:
https://github.com/griddynamics/rosetta

> > **Important**

Author: Use proper heading

Reviewer: fixed

>
> Cursor automatically detects and uses plugins installed in Claude Code. To avoid duplicate tools, commands, and context, install Rosetta in **Claude Code only**. Do not install the same plugin separately in both Claude Code and Cursor.

#### Standalone

1. Download `core-cursor-standalone-*.zip` from the [latest release](https://github.com/griddynamics/rosetta/releases/latest).
2. Extract the archive contents into your repository.
3. Verify you can see a file `.cursor/agents/architect.md`. Ensure there are no `.cursor/.cursor` folders.

</details>

<details>
<summary><b>GitHub Copilot</b></summary>

### GitHub Copilot

> Copilot runs in VS Code and JetBrains. The marketplace path is VS Code only; standalone works for both.

Author: Strange wording

Reviewer: rephrased

#### Marketplace (VS Code)

1. In VS Code settings, add `https://github.com/griddynamics/rosetta` to `chat.plugins.marketplaces`.
2. Open the Copilot chat panel, click the settings gear icon to open agent customizations.
3. Click **Browse Marketplaces**, then **install** for `rosetta`.

<img src="docs/images/vscode-add-marketplaces.png" alt="Add marketplaces to VS Code" width="710"/>

<img src="docs/images/vscode-open-customizations.png" alt="Open agent customizations" width="710"/>

<img src="docs/images/vscode-install-plugins.png" alt="Install plugins" width="710"/>

#### Standalone (VS Code and JetBrains)

> Use this option when marketplace installation is unavailable, such as in JetBrains Copilot or VS Code environments.

Author: JetBrains don't have one. If you wish to install -> have to use standalone, which VS Code will also see, thus you don't install in VS Code via marketplace (or you will have duplication!). Also standalone is the safest compatible option until Copilot in JetBrains catches up.

Reviewer: rephrased

1. Download `core-copilot-standalone-*.zip` from the [latest release](https://github.com/griddynamics/rosetta/releases/latest).
2. Extract the archive contents into your repository. If `.github/copilot-instructions.md` already exists, merge contents — Rosetta first, then the original content.
3. Verify you can see a file `.github/agents/architect.agent.md`. Ensure there are no `.github/.github` folders.

</details>

<details>
<summary><b>Codex</b></summary>

### Codex

> [!NOTE]
> Codex plugins currently support hooks, MCPs, and skills only (as of April 2026).

#### Standalone

1. Download `core-codex-*.zip` from the [latest release](https://github.com/griddynamics/rosetta/releases/latest).
2. Extract the archive contents into your repository.
3. Enable hooks:

   ```sh
   codex features enable hooks
   ```

</details>

## Step 2: Verify

Ask the agent:

```
What can you do, Rosetta?
```

It should use Rosetta MCP to retrieve agents, guardrails, and instructions:

> <img src="docs/images/Rosetta-ProperResponse1.png" alt="Rosetta proper response" width="355"/> <img src="docs/images/Rosetta-ProperResponse2.png" alt="Rosetta proper response" width="300"/>

Author: These are MCP screenshots

Author: Verify is another way:
1. check slash commands show flows like /coding-flow and /aqa-flow
2. check slash commands show skills like /debugging
3. What can you do, Rosetta? => new screenshots showing its behavior of loading CONTEXT.md/ARCHITECTURE.md/etc.

> [!CAUTION]
> You must receive prior approval from your manager and company to use Rosetta.

> [!WARNING]
> Use **Sonnet 4.6**, **GPT-5.4-medium**, **gemini-3.1-pro** or better models. Avoid Auto model selection.

> [!NOTE]
> This is pre-release, but it already works.
> There will be conflict if you have similar plugins installed: JUXT, Superpowers, GSD, AI-DevKit. Use the ones you have the most experience with.

## Step 1: Install Plugin

<details>
<summary><b>Claude Code</b></summary>

### Claude Code

#### Marketplace

```sh
claude plugin marketplace add griddynamics/rosetta
claude plugin install rosetta@rosetta
```
</details>

<details>
<summary><b>Cursor</b></summary>

### Cursor

#### Marketplace

> [!NOTE]
> To add the plugin you need to have the appropriate Cursor plans, such as Teams and Enterprise. 

To Import the Rosetta github repository to your team/compnay internal marketplace:
* Use the following repository: https://github.com/griddynamics/rosetta

For detailed setup instructions, see the Cursor documentation:
* https://cursor.com/docs/plugins#team-marketplaces

> Plugins installed in Claude Code are automatically available in Cursor.

Author: ```suggestion
**ALTERNATIVE**: Plugins installed in Claude Code are automatically available in Cursor.
```

#### Standalone (VS Code and JetBrains)

For JetBrains IDEs, use the standalone installation package. 
> [!NOTE]
> The standalone installation is also detected by VS Code, so installing Rosetta through the standalone and marketplace methods will result in duplicate tools, commands, and context.

1. Download `core-copilot-standalone-*.zip` from the [latest release](https://github.com/griddynamics/rosetta/releases/latest).
2. Extract the archive contents into your repository. If `.github/copilot-instructions.md` already exists, merge contents — Rosetta first, then the original content.
3. Verify you can see a file `.github/agents/architect.agent.md`. Ensure there are no `.github/.github` folders.

</details>

<details>
<summary><b>Codex</b></summary>

### Codex

> [!NOTE]
> Codex plugins currently support hooks, MCPs, and skills only (as of April 2026).

#### Standalone

1. Download `core-codex-*.zip` from the [latest release](https://github.com/griddynamics/rosetta/releases/latest).
2. Extract the archive contents into your repository.
3. Enable hooks:

   ```sh
   codex features enable hooks
   ```

</details>

## Step 2: Verify

Ask the agent:

```
What can you do, Rosetta?
```

> The agent will use Rosetta to retrieve agents, guardrails, and instructions:

Author: ```suggestion
The agent will follow Rosetta prompts and show Rosetta workflows and execute `self-help-flow` (see screenshots from different tools below):
```

<img src="docs/images/Rosetta-ProperResponse.png" alt="Rosetta proper response in Claude Code" width="710"/>

<img src="docs/images/Rosetta-ProperResponse-Copilot.png" alt="Rosetta proper response in GitHub Copilot" width="710"/>

> ## Updating

Author: ```suggestion
## Upgrading
```

> 

Author: ```suggestion

- Standalone upgrades require to redownload and replace files (install again).
- Marketplace plugins usually automatically upgrade.

```
